pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -42,10 +42,9 @@ import (
 // These are intentionally pre-computed based on events such that lookups are efficient.
 type AmbientIndex struct {
 	mu sync.RWMutex
-	// byService indexes by network/Service (virtual) *IP address*. A given Service may have multiple IPs, thus
-	// multiple entries in the map.
-	// A given IP can map to many workloads associated, indexed by workload uid.
-	byService map[networkAddress]map[string]*model.WorkloadInfo
+	// byService indexes by Service namespaced hostname. A given Service can map to
+	// many workloads associated, indexed by workload uid.
+	byService map[string]map[string]*model.WorkloadInfo
```

Comment:
I am not opposed with this change, just a doubt fort k8s service, how could hostname un-unique

Reply:
Not sure what you're asking here. Are you asking if hostname is unique?

Reply:
yes, host name for kube registry is unique